### PR TITLE
[PBW-4447][VTC]Pass asset's delivery type on setting video plugin's URL

### PR DESCRIPTION
**Pull request by mityaha**

### CODE REVIEW

File: src/main/js/main_html5.js [hunk 1]
```diff
@@ -267,7 +268,7 @@ require("../../../html5-common/js/utils/environment.js");
           _currentUrl = _currentUrl + (/\?/.test(_currentUrl) ? "&" : "?") + "_=" + getRandomString();
         }
 
-        isM3u8 = (_currentUrl.toLowerCase().indexOf("m3u8") > 0);
+        isM3u8 = (_currentUrl.toLowerCase().indexOf("m3u8") > 0); // XXX this should go away, as we have encoding now
```
Comment (angieisrani) [inline src/main/js/main_html5.js]: true, this can go; why not do it with this change?

File: src/osmf/js/osmf_flash.js [hunk 2]
```diff
@@ -413,7 +414,7 @@ require("../../../html5-common/js/utils/constants.js");
       // Pause the video
       this.pause();
       // Reset the source
-      this.setVideoUrl('');
+      this.setVideoUrl('', '');
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: Why do you need to pass an empty string here?  The parameter is not being used so it's okay if it's null or undefined (since '' is not more accurate anyways)

File: src/template.js [hunk 3]
```diff
@@ -175,9 +175,10 @@ require("../html5-common/js/utils/constants.js");
      * @public
      * @method TemplateVideoWrapper#setVideoUrl
      * @param {string} url The new url to insert into the video element's src attribute
+     * @param {string} encoding The encoding of video stream
```
Comment (angieisrani) [inline src/template.js]: Let's also specify that possible encodings are defined in OO.VIDEO.ENCODING

File: tests/unit/main_html5/wrapper-api-tests.js [hunk 4]
```diff
@@ -47,35 +47,35 @@ describe('main_html5 wrapper tests', function () {
   });
 
   it('should set the video url and return true', function(){
-    var returns = wrapper.setVideoUrl("url");
+    var returns = wrapper.setVideoUrl("url", "mp4");
```
Comment (angieisrani) [inline tests/unit/main_html5/wrapper-api-tests.js]: Are these changes needed?  Do they fail without the parameter?

Comment (mityaha) [inline tests/unit/main_html5/wrapper-api-tests.js]: Removed

File: src/main/js/main_html5.js (hunk 1 above)
Comment (mityaha) [inline src/main/js/main_html5.js]: Done

**Comment by angieisrani** [comment]
LGTM - will merge all commits together when all are ready
